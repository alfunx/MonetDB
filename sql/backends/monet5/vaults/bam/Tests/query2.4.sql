SELECT qname, flag, rname, pos, mapq, cigar, rnext, pnext, tlen, seq, qual
FROM bam.alignments_1
WHERE qname IN (
    SELECT qname
    FROM (
        SELECT qname, SUM(segm_unma), SUM(seco_alig), COUNT(*)
        FROM (
            SELECT qname, bam.bam_flag(flag, 'seco_alig') AS seco_alig, bam.bam_flag(flag, 'segm_unma') AS segm_unma, 
                bam.bam_flag(flag, 'firs_segm') AS firs_segm
            FROM bam.alignments_1
            WHERE bam.bam_flag(flag, 'firs_segm') <> bam.bam_flag(flag, 'last_segm')
        ) AS sub
        GROUP BY qname, firs_segm
        HAVING SUM(segm_unma) < COUNT(*)
           AND (COUNT(*) - SUM(seco_alig)) <> 1
    ) AS qnames
);


SELECT qname, flag, rname, pos, mapq, cigar, rnext, pnext, tlen, seq, qual
FROM bam.unpaired_all_alignments_3
WHERE qname IN (
    SELECT qname
    FROM (
        SELECT qname, SUM(segm_unma), SUM(seco_alig), COUNT(*)
        FROM (
            SELECT qname, bam.bam_flag(flag, 'seco_alig') AS seco_alig, bam.bam_flag(flag, 'segm_unma') AS segm_unma, 
                bam.bam_flag(flag, 'firs_segm') AS firs_segm
            FROM bam.unpaired_all_alignments_3
            WHERE bam.bam_flag(flag, 'firs_segm') <> bam.bam_flag(flag, 'last_segm')
        ) AS sub
        GROUP BY qname, firs_segm
        HAVING SUM(segm_unma) < COUNT(*)
           AND (COUNT(*) - SUM(seco_alig)) <> 1
    ) AS qnames
);


