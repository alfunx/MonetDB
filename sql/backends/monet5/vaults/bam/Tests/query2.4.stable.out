stdout of test 'query2.4` in directory 'sql/backends/monet5/vaults/bam` itself:


# 17:36:06 >  
# 17:36:06 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=33676" "--set" "mapi_usock=/var/tmp/mtest-11149/.s.monetdb.33676" "--set" "monet_prompt=" "--forcemito" "--set" "mal_listing=2" "--dbpath=/home/robin/MonetDB/PREFIX/var/MonetDB/mTests_sql_backends_monet5_bam" "--set" "mal_listing=0"
# 17:36:06 >  

# MonetDB 5 server v11.18.0
# This is an unreleased version
# Serving database 'mTests_sql_backends_monet5_bam', using 4 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs dynamically linked
# Found 7.333 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2015 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://robin-xps13:33676/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-11149/.s.monetdb.33676
# MonetDB/SQL module loaded

Ready.

# 17:36:06 >  
# 17:36:06 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-11149" "--port=33676"
# 17:36:06 >  

#SELECT qname, flag, rname, pos, mapq, cigar, rnext, pnext, tlen, seq, qual
#FROM bam.alignments_1
#WHERE qname IN (
#    SELECT qname
#    FROM (
#        SELECT qname, SUM(segm_unma), SUM(seco_alig), COUNT(*)
#        FROM (
#            SELECT qname, bam.bam_flag(flag, 'seco_alig') AS seco_alig, bam.bam_flag(flag, 'segm_unma') AS segm_unma, 
#                bam.bam_flag(flag, 'firs_segm') AS firs_segm
#            FROM bam.alignments_1
#            WHERE bam.bam_flag(flag, 'firs_segm') <> bam.bam_flag(flag, 'last_segm')
% bam.alignments_1,	bam.alignments_1,	bam.alignments_1,	bam.alignments_1,	bam.alignments_1,	bam.alignments_1,	bam.alignments_1,	bam.alignments_1,	bam.alignments_1,	bam.alignments_1,	bam.alignments_1 # table_name
% qname,	flag,	rname,	pos,	mapq,	cigar,	rnext,	pnext,	tlen,	seq,	qual # name
% clob,	smallint,	clob,	int,	smallint,	clob,	clob,	int,	int,	clob,	clob # type
% 0,	1,	0,	1,	1,	0,	0,	1,	1,	0,	0 # length
#SELECT qname, flag, rname, pos, mapq, cigar, rnext, pnext, tlen, seq, qual
#FROM bam.unpaired_all_alignments_3
#WHERE qname IN (
#    SELECT qname
#    FROM (
#        SELECT qname, SUM(segm_unma), SUM(seco_alig), COUNT(*)
#        FROM (
#            SELECT qname, bam.bam_flag(flag, 'seco_alig') AS seco_alig, bam.bam_flag(flag, 'segm_unma') AS segm_unma, 
#                bam.bam_flag(flag, 'firs_segm') AS firs_segm
#            FROM bam.unpaired_all_alignments_3
#            WHERE bam.bam_flag(flag, 'firs_segm') <> bam.bam_flag(flag, 'last_segm')
% .unpaired_all_alignments_3,	.unpaired_all_alignments_3,	.unpaired_all_alignments_3,	.unpaired_all_alignments_3,	.unpaired_all_alignments_3,	.unpaired_all_alignments_3,	.unpaired_all_alignments_3,	.unpaired_all_alignments_3,	.unpaired_all_alignments_3,	.unpaired_all_alignments_3,	.unpaired_all_alignments_3 # table_name
% qname,	flag,	rname,	pos,	mapq,	cigar,	rnext,	pnext,	tlen,	seq,	qual # name
% clob,	smallint,	clob,	int,	smallint,	clob,	clob,	int,	int,	clob,	clob # type
% 0,	1,	0,	1,	1,	0,	0,	1,	1,	0,	0 # length

# 17:36:06 >  
# 17:36:06 >  "Done."
# 17:36:06 >  
